docs/resources/stream_processor.md

```diff
@@ -0,0 +1,97 @@
+
+`mongodbatlas_stream_processor` provides a Stream Processor resource. The resource lets you create, delete, import, start and stop a stream processor in a stream instance.
+
+## Example Usages
+
+```terraform
+resource "mongodbatlas_stream_instance" "example" {
+  project_id    = var.project_id
+  instance_name = "InstanceName"
+  data_process_region = {
+    region         = "VIRGINIA_USA"
+    cloud_provider = "AWS"
+  }
+}
+
+resource "mongodbatlas_stream_connection" "example-sample" {
+  project_id      = var.project_id
+  instance_name   = mongodbatlas_stream_instance.example.instance_name
+  connection_name = "sample_stream_solar"
+  type            = "Sample"
+}
+
+resource "mongodbatlas_stream_processor" "stream-processor-example" {
+  project_id     = var.project_id
+  instance_name  = mongodbatlas_stream_instance.example.instance_name
+  processor_name = "processorName"
+  pipeline       = "[{\"$source\":{\"connectionName\":\"sample_stream_solar\"}},{\"$emit\":{\"connectionName\":\"__testLog\"}}]"
+  state          = "STARTED"
+}
+
+data "mongodbatlas_stream_processors" "example-stream-processors" {
+  project_id    = var.project_id
+  instance_name = mongodbatlas_stream_instance.example.instance_name
+}
+
+data "mongodbatlas_stream_processor" "example-stream-processor" {
+  project_id     = var.project_id
+  instance_name  = mongodbatlas_stream_instance.example.instance_name
+  processor_name = mongodbatlas_stream_processor.stream-processor-example.processor_name
+}
+
+# example making use of data sources
+output "stream_processors_state" {
+  value = data.mongodbatlas_stream_processor.example-stream-processor.state
+}
+
+output "stream_processors_results" {
+  value = data.mongodbatlas_stream_processors.example-stream-processors.results
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `instance_name` (String) Human-readable label that identifies the stream instance.
+- `pipeline` (String) Stream aggregation pipeline you want to apply to your streaming data.
+- `processor_name` (String) Human-readable label that identifies the stream processor.
+- `project_id` (String) Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.
+
+**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.
+
+### Optional
+
+- `options` (Attributes) Optional configuration for the stream processor. (see [below for nested schema](#nestedatt--options))
+- `state` (String) The state of the stream processor. Used to start or stop the Stream Processor. Valid values are `CREATED`, `STARTED` or `STOPPED`  When a Stream Processor is created without specifying the state, it will default to `CREATED` state.
```

Comment:
All documentation is autogenerated except for this attribute. Autogenerated description of this is only:
```
- `state` (String) The state of the stream processor.
```